engine/classes/ElggDiskFilestore.php

```diff
@@ -68,7 +68,7 @@ public function open(\ElggFile $file, $mode) {
 
 		$path = substr($fullname, 0, $ls);
 
-		if (($mode != 'write') && (!file_exists($fullname))) {
+		if (($mode === 'open') && (!file_exists($fullname))) {
```

Comment:
`@param string    $mode read, write, or append.` 
Where does `open` come from? I think we should validate the modes first.